lesson_05/script.js

```diff
@@ -6,7 +6,9 @@ const data = [
   { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
 ];
 
-const result = new Promise(function (resolve, reject) {
+const promise = new Promise((resolve) => {
+  resolve(data);
+}).then(() => {
   for (let i = 0; i < data.length; i++) {
     const li = document.createElement("li");
     const a = document.createElement("a");
```

Comment:
Now you are referring to the value of a global variable, not the resolved value.
The "data" in the loop process refers to the global variable "data".

Specify the .then argument to receive the resolved value.
````
}).then((data) => {
  for (let i = 0; i < data.length; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
````

Reply:
This is a recommended article about resolve and then arguments.
The resolve and then arguments do not necessarily have to be the same.

https://tcd-theme.com/2021/09/javascript-promise1.html?gclid=Cj0KCQiAmpyRBhC-ARIsABs2EAqpROtKKXtYEuMB2qKEApGWJTxEHAuHNx0SgbMlw4da_AVNFJzQEkMaAldFEALw_wcB